src/send-message.ts

```diff
@@ -0,0 +1,174 @@
+      return validValues.includes(input as any)
+        ? Ok(input as DestinationKind)
+        : Err(errorMessage);
+    });
+}
+
+function getDestinationDetails(): Result<DestinationDetails, string> {
+  return getDestinationKindInputFromJob()
+    .andThen((destinationKind) => {
+      return getMandatoryInputFromJob("to").map((destination) => {
+        return { kind: destinationKind, destination };
+      });
+    })
+    .andThen(parseDestinationDetails);
+}
+
+function parseDestinationDetails({
+  kind,
+  destination,
+}: {
+  kind: DestinationKind;
+  destination: string;
+}): Result<DestinationDetails, string> {
+  switch (kind) {
+    case DestinationKind.Private: {
+      return Ok({
+        kind,
+        destination: parsePrivateMessageDestinations(destination),
+      });
+    }
+
+    case DestinationKind.Stream: {
+      return getMandatoryInputFromJob("topic")
+        .map((topic) => {
+          return {
+            kind,
+            topic,
+            destination: parseStreamDestination(destination),
+          };
+        })
+        .mapErr(() => 'topic is mandatory when type is "stream"');
+    }
+
+    default: {
+      assertExhaustiveSwitch(kind);
+    }
+  }
+}
+
+function parsePrivateMessageDestinations(
+  input: string
+): PrivateDestination["destination"] {
+  let rawDestinations = input.split(",");
+  return allNumeric(rawDestinations)
+    ? rawDestinations.map((item: string) => Number.parseInt(item))
+    : rawDestinations;
+}
+
+function parseStreamDestination(
+  input: string
+): StreamDestination["destination"] {
+  return allNumeric(input) ? Number.parseInt(input) : input;
+}
+
+async function getZulipClient(): Promise<Result<Client, string>> {
+  return new AsyncResultWrapper(
+    Result.all(
+      getMandatoryInputFromJob("api-key"),
+      getMandatoryInputFromJob("email"),
+      getMandatoryInputFromJob("organization-url")
+    )
+  )
+    .flatMap(
+      ([apiKey, username, realm]) =>
+        new AsyncResultWrapper(
+          Result.wrapAsync(async () =>
+            zulipInit({
+              apiKey,
+              username,
+              realm,
+            })
+          )
+        )
+    )
+    .resolve();
+}
+
+async function postMessageFromJobInputs(): Promise<Result<string, string>> {
+  const topic = getInput("topic", { required: false });
+  return new AsyncResultWrapper(
+    Result.all(
+      getMandatoryInputFromJob("content"),
+      getDestinationDetails(),
+      await getZulipClient()
+    )
+  )
+    .flatMap(([content, destination, client]) => {
+      const parameters = {
+        to: destination.destination,
+        kind: destination.kind,
+        topic,
+        content,
+      };
+      return new AsyncResultWrapper(
+        Result.wrapAsync(async () => client.messages.send(parameters))
+      );
+    })
+    .flatMap((response: any) => {
+      return response.result === "success"
+        ? Ok(`Message successfully send with id: ${response.id as string}`)
```

Comment:
```suggestion
        ? Ok(`Message successfully sent with id: ${response.id as string}`)

Reply:
String was preserved from original JS implementation, L57 on `main` has the same typo :) This is a good place to fix that, thanks!

Reply:
Hah.  Well, I think for this commit we probably don't have to preserve bug-for-bug compatibility. :)